### PR TITLE
add KeepEventAuthor

### PR DESCRIPTION
**Pull request by camhashemi**
@ryanpbrewster @leogrim 

per @DerekBurch and I's conference, each keep event has an `author`, either user, slack, twitter, or email. Created a new class rather than stretching `BasicAuthor` thin to include emails.

### CODE REVIEW

File: kifi-backend/modules/common/app/com/keepit/social/BasicUser.scala [hunk 1]
```diff
@@ -75,7 +79,13 @@ object BasicUser {
...
+    chooseImage(userId.id)
+  }
+
+  def defaultImageForEmail(email: String): String = chooseImage(MurmurHash3.stringHash(email))
+
+  private def chooseImage(randomLong: Long): String = {
+    val variant = (randomLong % 42).toInt
     val choice = (variant: @switch) match {
       case 0 => "bat"
       case 1 => "bear"
```
Comment (camhashemi) [inline kifi-backend/modules/common/app/com/keepit/social/BasicUser.scala:91]: @andrewconner opening this up as a LHF for images on email senders

**Comment by ryanpbrewster** [comment]
TBH, my only feedback on this is that I think you should just extend `Author/BasicAuthor` instead of introducing a new type. I am not convinced my gut feeling is right, but that's where I lean.

**Comment by ryanpbrewster** [comment]
Had another look at what would be necessary to add `AuthorKind.Email` and it's actually a bit of a pain because of how I structured things. :(

I still think we should do it, though.

**Comment by camhashemi** [comment]
It DRYs things up and clients are familiar with the `BasicAuthor` object (cc @DerekBurch), it just makes the class less specific. e.g. should `BasicAuthor.picture, BasicAuthor.name` be `Option[String]` now, or set them to garbage values for `BasicAuthor.EmailUser`? either way, the class becomes less useful, albeit more useable. I'll go ahead with it for now, we can always break off into a new class like we have here.

**Comment by ryanpbrewster** [comment]
:+1: works for me

File: kifi-backend/modules/common/app/com/keepit/social/Author.scala [hunk 2]
```diff
@@ -94,16 +96,26 @@ object BasicAuthor {
...
     )
   }
+  def fromNonUser(nonUser: BasicNonUser): BasicAuthor = nonUser.kind match {
+    case NonUserKinds.email =>
+      EmailUser(
+        nonUser.id,
+        nonUser.firstName.getOrElse(nonUser.id),
+        BasicUser.defaultImageForEmail(nonUser.id)
+      )
+  }
```
Comment (camhashemi) [inline kifi-backend/modules/common/app/com/keepit/social/Author.scala:106]: @ryanpbrewster updated

**Pull request merged**